### PR TITLE
update command for 2018 some more

### PR DESCRIPTION
**Pull request by ariovistus**

### CODE REVIEW

File: wpilib/wpilib/command/command.py [hunk 1]
```diff
@@ -49,18 +49,21 @@ def __init__(self, name=None, timeout=None):
...
+            name = self.__class__.__name__
+
+        self.setName(name)
 
         # The time (in seconds) before this command "times out" (or None if no
         # timeout)
         if timeout is not None and timeout < 0:
             raise ValueError("Timeout must not be negative")
+        if timeout is None:
+            timeout = -1.0
```
Comment (auscompgeek) [inline wpilib/wpilib/command/command.py]: Hmm. Is there a reason for adding this?

File: wpilib/wpilib/command/subsystem.py [hunk 2]
```diff
@@ -138,36 +133,36 @@ def getCurrentCommand(self):
...
-        
-        :returns: the name of this subsystem
+        :returns: the current command name
+        """
+        currentCommand = self.getCurrentCommand()
+        if currentCommand is not None:
+            return currentCommand.getName()
+        return ""
+
+    def addChild(self, child, name=None):
```
Comment (auscompgeek) [inline wpilib/wpilib/command/subsystem.py]: This doesn't match the order of the optional `name` upstream. This is a new API, so we could deviate from upstream I suppose. Do we want to here, @virtuald?

File: wpilib/wpilib/command/subsystem.py [hunk 3]
```diff
@@ -138,36 +133,36 @@ def getCurrentCommand(self):
...
-                self.hasDefaultEntry.setBoolean(False)
-
-            if self.currentCommand is not None:
-                self.hasCommandEntry.setBoolean(True)
-                self.commandEntry.setString(self.currentCommand.getName())
-            else:
-                self.hasCommandEntry.setBoolean(False)
+        Associate a :class:`.Sendable` with this Subsystem.
+        Update the child's name if provided
+        :param child: sendable
```
Comment (auscompgeek) [inline wpilib/wpilib/command/subsystem.py]: Add a blank line before this.

Comment (Twinters007) [reply wpilib/wpilib/command/subsystem.py]: And capitalize Sendable

Comment (auscompgeek) [reply wpilib/wpilib/command/subsystem.py]: @Twinters007 File a bug upstream if you care.

Comment (Twinters007) [reply wpilib/wpilib/command/subsystem.py]: We can differ on capitalization, no?

Comment (auscompgeek) [reply wpilib/wpilib/command/subsystem.py]: I don't see any reason to.

Comment (Twinters007) [reply wpilib/wpilib/command/subsystem.py]: Looks better in the docs imo.

Comment (auscompgeek) [reply wpilib/wpilib/command/subsystem.py]: Inconsistency is bad, and _doesn't_ look good.

Comment (virtuald) [reply wpilib/wpilib/command/subsystem.py]: I think change it to be Right here _and_ send a PR upstream.

File: wpilib/wpilib/command/subsystem.py (hunk 2 above)
Comment (virtuald) [reply wpilib/wpilib/command/subsystem.py]: Optional arguments have to go last, so deviation is fine.

File: wpilib/wpilib/command/subsystem.py (hunk 3 above)
Comment (auscompgeek) [reply wpilib/wpilib/command/subsystem.py]: We should also change the rest of them to be consistent then.

File: wpilib/wpilib/command/command.py (hunk 1 above)
Comment (ariovistus) [reply wpilib/wpilib/command/command.py]: yes, in the java wpilib the default value is -1.0, and other methods in this class expect it to be -1.0 and break if it is None. There are tests that exercise this.

**Comment by ariovistus** [comment]
blank line added, could other changes be done in a separate PR please? 

\<bikeshedding>actually, if you were to capitalize it, you're just reporting the type, so shouldn't it be

```
The :class:`.Sendable`
```
\</bikeshedding>

**Comment by Twinters007** [comment]
I wanted you to capitalize it because it's like a sentence and should start with caps (imo)

**Comment by ariovistus** [comment]
making that a consistently followed rule sounds like an even bigger task

File: wpilib/wpilib/command/command.py (hunk 1 above)
Comment (Twinters007) [reply wpilib/wpilib/command/command.py]: 2 lines above you say timeout cannot be negative, yet here you set it to negative...Why not set it to -1 if it's negative?

**Comment by Twinters007** [comment]
I know I've been doing it... ¯\\\_(ツ)\_/¯

**Comment by ariovistus** [comment]
are these conventions being documented anywhere? seems like an explicit checklist would be handy for new contributors.

**Comment by Twinters007** [comment]
Eh not really. Notice how we also fight over whitespace a lot

**Comment by virtuald** [comment]
You could just document it by changing the robotpy-docs and see if the PR gets merged.

**Pull request merged**